### PR TITLE
feat(Storybook): Add integrated test runner

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

This PR adds ability o run unit tests related to given Storybook page, automatically or for demand.
For isolation tests re run in a separate thread. It's possible to abort testing in progress, results are logged to QtCreator (when Storybook is run via QtCreator). Settings switch is added to enable/disable automatic tests re-run.

Closes: #12331 

### Affected areas
`Storybook`

### Screenshot of functionality (including design for comparison)

[Kazam_screencast_00046.webm](https://github.com/status-im/status-desktop/assets/20650004/7451dfda-b5de-47e8-b666-78f43cb7fb29)

### CODE REVIEW

File: ui/StatusQ/src/QClipboardProxy.cpp [hunk 1]
```diff
@@ -36,7 +36,8 @@ bool QClipboardProxy::hasHtml() const
 
 QString QClipboardProxy::html() const
 {
-    return m_clipboard->mimeData()->html();
+    auto mimeData = m_clipboard->mimeData();
+    return mimeData ? mimeData->html() : QLatin1String();
```
Comment (caybro) [inline ui/StatusQ/src/QClipboardProxy.cpp:40]:
```
return mimeData ? mimeData->html() : {};
```

File: storybook/testsrunner.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    QProcess testRunnerProcess;
+    testRunnerProcess.start(m_testRunnerExecutablePath, arguments);
+    testRunnerProcess.waitForFinished();
+
+    if (testRunnerProcess.exitCode())
+        return 0;
+
+    QByteArray functions = testRunnerProcess.readAllStandardError();
+    return functions.count('\n');
```
Comment (caybro) [inline storybook/testsrunner.cpp:27]: But this sounds problematic to anything else than Linux :)

File: storybook/testsrunner.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+QObject* TestsRunner::runTests(const QString& fileName)
+{
+    QStringList arguments;
+    arguments << "-platform" << "offscreen";
+    arguments << "-input" << m_testsBasePath + QDir::separator() + fileName;
```
Comment (caybro) [inline storybook/testsrunner.cpp:34]: Ditto, use `/`

**Review by caybro — APPROVED** [review]
Really nice work, didn't test but the code looks good, some minor stuff only

File: ui/StatusQ/src/QClipboardProxy.cpp (hunk 1 above)
Comment (micieslak) [reply ui/StatusQ/src/QClipboardProxy.cpp:40]: Initializer list cannot be used on the right hand side of operator `:`, but `QString{}` seems to be a good option here.

File: storybook/testsrunner.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    : QObject{parent}, m_testRunnerExecutablePath{testRunnerExecutablePath},
+      m_testsBasePath{testsBasePath}
+{
+}
+
+int TestsRunner::testsCount(const QString& fileName)
+{
+    QStringList arguments;
+    arguments << "-functions";
+    arguments << "-input" << m_testsBasePath + QDir::separator() + fileName;
```
[the remark this answers is not in the document]
Comment (micieslak) [reply storybook/testsrunner.cpp:17]: I wrapped literals with `QStringLiteral`, doing same thing for "/" would result in longer code here so I'm leaving `QDir::separator()` here.

File: storybook/testsrunner.cpp (hunk 2 above)
Comment (micieslak) [reply storybook/testsrunner.cpp:30]: Yeah, good point. To fix that I started the process via `open` with `QIODevice::Text` mode. According to https://doc.qt.io/qt-6/qiodevicebase.html#OpenModeFlag-enum it should cause translation of native newlines to `\n`.

File: ui/StatusQ/src/QClipboardProxy.cpp (hunk 1 above)
Comment (caybro) [reply ui/StatusQ/src/QClipboardProxy.cpp]: Oh, TIL

File: storybook/testsrunner.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (caybro) [reply storybook/testsrunner.cpp]: Yeah sure, not a big deal

File: storybook/testsrunner.cpp (hunk 2 above)
Comment (caybro) [reply storybook/testsrunner.cpp:30]: Yup, good idea!

**Review by alexjba — APPROVED** [review]
Beautiful, thank you! 🍻

**Pull request merged**